Bottom: 9b6c62dd378c48e2a42ed7b9d4712eaa7a69c668
Top:    df9566f2e48f4bbf2d90fac4beba92673bf52f60
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2012-03-22 19:39:19 -0400

Refresh of spl

---

diff --git a/arch/arm/cpu/arm926ejs/lpc313x/config.mk b/arch/arm/cpu/arm926ejs/lpc313x/config.mk
index 85e9f1f..2b0ea47 100644
--- a/arch/arm/cpu/arm926ejs/lpc313x/config.mk
+++ b/arch/arm/cpu/arm926ejs/lpc313x/config.mk
@@ -32,3 +32,7 @@ PLATFORM_CPPFLAGS += -march=armv5te
 PLATFORM_RELFLAGS +=$(call cc-option,-mshort-load-bytes,$(call cc-option,-malignment-traps,))
 
 USE_PRIVATE_LIBGCC := yes
+
+ifndef CONFIG_SPL_BUILD
+ALL-y	+= $(obj)u-boot.img
+endif
diff --git a/include/configs/ea3131.h b/include/configs/ea3131.h
index 47f5231..ff2b028 100644
--- a/include/configs/ea3131.h
+++ b/include/configs/ea3131.h
@@ -168,9 +168,10 @@
 #define CONFIG_BOOTDELAY		1
 #define CONFIG_EXTRA_ENV_SETTINGS	\
 "loadaddr=0x31000000\0" \
-"rd_addr=0x32000000\0" \
+"dt_addr=0x32000000\0" \
+"rd_addr=0x32100000\0" \
 "usbtty=cdc_acm\0" \
-"ramargs=setenv bootargs console=ttyS0,115200n8 root=/dev/mmcblk0p3 init=/etc/preinit rw ip=182.168.1.180 loglevel=7\0" \
+"ramargs=setenv bootargs console=ttyS0,115200n8 root=/dev/mmcblk0p3 init=/etc/preinit rw loglevel=7\0" \
 "serverip=192.168.1.48\0" \
 "nfsargs=setenv bootargs console=ttyS0,115200n8 root=/dev/nfs init=/etc/preinit rw nfsroot=${serverip}:${rootpath} ip=dhcp loglevel=7\0" \
 "rootpath=/tftpboot/arm\0" \
@@ -179,7 +180,7 @@
 "net_boot=dhcp; run nfsargs; bootm $(loadaddr)\0" \
 "spi_boot= sf probe 0 0 0; sf read $(loadaddr) 0x42000 0x200000; run nfsargs; bootm $(loadaddr)\0" \
 "nand_boot= nand read $(loadaddr) 0x80000 0x200000; run ramargs; bootm $(loadaddr)\0" \
-"sdmmc_boot= mmc init; fatload mmc 0 $(loadaddr) $(bootfile); run ramargs; bootm $(loadaddr)\0" \
+"sdmmc_boot= mmc init; fatload mmc 0 $(loadaddr) $(bootfile); fatload mmc 0 $(dt_addr) ea3131.dtb; run ramargs; bootm $(loadaddr) - $(dt_addr)\0" \
 "usbdfu_boot= usbpoll $(loadaddr); run nfsargs; bootm $(loadaddr)\0" \
 "usb_boot= usb start; fatload usb 0 $(loadaddr) $(bootfile); run nfsargs; bootm $(loadaddr)\0" \
 "uart_boot= loady; run nfsargs; bootm $(loadaddr)\0" \
